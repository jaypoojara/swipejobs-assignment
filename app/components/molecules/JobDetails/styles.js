import { StyleSheet } from 'react-native';
import colors from '../../../themes/colors';
import fonts from '../../../themes/fonts';
export default StyleSheet.create({
  container: {
    height: '85%',
    backgroundColor: colors.backGround
  },
  scrollContainerWrapper: {
    backgroundColor: 'white',
    flex: 1,
    borderRadius: 8,
    margin: 20
  },
  itemContainer: {
    flex: 1,
    overflow: 'hidden',
    borderRadius: 8
  },
  companyContainer: {
    marginHorizontal: 10,
    marginVertical: 10
  },
  jobTitleText: {
    ...fonts.style.overSizeBold
  },
  companyNameText: {
    fontSize: 16,
    marginTop: 10
  },
  distanceContainer: {
    flexDirection: 'row',
    paddingHorizontal: 10,
    justifyContent: 'space-between',
    backgroundColor: colors.banner,
    paddingVertical: 10
  },
  distanceText: {
    ...fonts.style.mediumBold
  },
  milesText: {
    ...fonts.style.extraOverSizeBold,
    color: colors.white,
    marginTop: 5
  },
  hourlyRateText: {
    ...fonts.style.mediumBold,
    textAlign: 'right'
  },
  dollarContainer: {
    flexDirection: 'row',
    alignItems: 'flex-start',
    marginTop: 5
  },
  dollarText: {
    ...fonts.style.overSizeBold,
    color: colors.white,
    textAlign: 'right',
    lineHeight: 30
  },
  priceText: {
    ...fonts.style.extraOverSizeBold,
    color: colors.white,
    textAlign: 'right'
  },
  detailContainer: {
    marginHorizontal: 15,
    paddingVertical: 10,
    borderBottomWidth: 1,
    flexDirection: 'row',
    alignItems: 'center'
  },
  detailText: {
    ...fonts.style.extraLargeBold,
    color: colors.black
  },
  slotText: {
    fontSize: fonts.size.extraLarge,
    color: colors.black
  },
  ML: { marginLeft: 15 },
  extraLarge: {
    fontSize: fonts.size.extraLarge,
    color: colors.black
  },
  helperText: {
    fontSize: fonts.size.large,
    color: colors.black,
    marginTop: 5
  },
  ML10: {
    marginLeft: 10
  },
  border: {
    borderRadius: 8,
    flex: 1
  },
  image: {
    height: 150
  },
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: colors.white,
    paddingVertical: 10,
    paddingHorizontal: 10,
    borderBottomLeftRadius: 8,
    borderBottomRightRadius: 8
  }
});
