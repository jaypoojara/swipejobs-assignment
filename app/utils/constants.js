export const NOTIFICATION_TYPE = {
  DEFAULT: 0,
  SUCCESS: 1,
  ERROR: 2,
  WARNING: 3,
  INFO: 4
};

export const WORKER_ID = '7f90df6e-b832-44e2-b624-3143d428001f';
