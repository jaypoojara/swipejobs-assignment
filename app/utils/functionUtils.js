import moment from 'moment';
import momentTz from 'moment-timezone';

export const getDuration = (startDate, endDate) => {
  const days = moment(startDate)
    .local()
    .format('MMM DD, ddd HH:mm A');
  const endDateTemp = moment(endDate)
    .local()
    .format('HH:mm A');
  const timeZone = momentTz.tz.guess();
  const timeZoneAbr = momentTz.tz(timeZone).zoneAbbr();
  return `${days} - ${endDateTemp} ${timeZoneAbr}`;
};
