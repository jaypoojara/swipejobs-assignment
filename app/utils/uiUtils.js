import { showMessage } from 'react-native-flash-message';
import { NOTIFICATION_TYPE } from './constants';
export function showMessageToUser({ title, message, notificationType }) {
  let messageType;
  switch (notificationType) {
    case NOTIFICATION_TYPE.DEFAULT:
      messageType = 'default';
      break;
    case NOTIFICATION_TYPE.INFO:
      messageType = 'info';
      break;
    case NOTIFICATION_TYPE.SUCCESS:
      messageType = 'success';
      break;
    case NOTIFICATION_TYPE.ERROR:
      messageType = 'danger';
      break;
    case NOTIFICATION_TYPE.WARNING:
      messageType = 'warning';
      break;
    default:
      messageType = 'none';
  }
  setTimeout(() => {
    showMessage({
      message: title,
      description: message,
      type: messageType
    });
  }, 100);
}
