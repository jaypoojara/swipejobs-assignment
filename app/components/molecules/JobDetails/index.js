import React from 'react';
import { View, Image, ScrollView } from 'react-native';
import { FontAwesome5 } from '@expo/vector-icons';
import PropTypes from 'prop-types';
import colors from '../../../themes/colors';
import T from '../../atoms/T';
import ButtonComponent from '../../atoms/Button';
import { getDuration } from '../../../utils/functionUtils';
import styles from './styles';

const JobDetails = ({ item: jobDetails, acceptRejectJobs }) => (
  <View style={styles.container}>
    <View style={styles.scrollContainerWrapper}>
      <ScrollView style={styles.border}>
        <View style={styles.itemContainer}>
          <Image
            style={styles.image}
            source={{
              uri: jobDetails.jobTitle.imageUrl
            }}
          />
          <View style={styles.companyContainer}>
            <T style={styles.jobTitleText} text={jobDetails.jobTitle.name} />
            <T style={styles.companyNameText} text={jobDetails.company.name} />
          </View>
          <View style={styles.distanceContainer}>
            <View>
              <T style={styles.distanceText} id="distance" />
              <T
                style={styles.milesText}
                text={`${jobDetails.milesToTravel.toFixed(2)} miles`}
              />
            </View>
            <View>
              <T style={styles.hourlyRateText} id="hourly_text" />
              <View style={styles.dollarContainer}>
                <T style={styles.dollarText} text="$ " />
                <T
                  style={styles.priceText}
                  text={`${(jobDetails.wagePerHourInCents / 100).toFixed(2)}`}
                />
              </View>
            </View>
          </View>
          <View style={styles.detailContainer}>
            <View>
              <FontAwesome5 name="calendar-alt" size={24} color="black" />
            </View>
            <View style={styles.ML}>
              <T id="shift_text" style={styles.detailText} />
              {jobDetails.shifts.length > 0 &&
                jobDetails.shifts.map((shift, index) => (
                  <T
                    key={`${shift.startDate}-${shift.endDate}-${index}`}
                    style={styles.slotText}
                    text={`${getDuration(
                      shift.startDate,
                      shift.endDate
                    ).toUpperCase()}`}
                  />
                ))}
            </View>
          </View>

          <View style={styles.detailContainer}>
            <View>
              <FontAwesome5 name="location-arrow" size={24} color="black" />
            </View>
            <View style={styles.ML}>
              <T id="location" style={styles.detailText} />
              <T
                style={styles.extraLarge}
                text={`${jobDetails.company.address.formattedAddress}`}
              />
              <T
                text={`${jobDetails.milesToTravel.toFixed(
                  2
                )} miles from your job search location`}
                style={styles.helperText}
              />
            </View>
            <View style={styles.ML10}>
              <FontAwesome5 name="chevron-right" size={24} color="black" />
            </View>
          </View>
          {jobDetails.requirements && jobDetails.requirements?.length > 0 && (
            <View style={styles.detailContainer}>
              <View>
                <FontAwesome5 name="hammer" size={24} color="black" />
              </View>
              <View style={styles.ML}>
                <T id="requirement" style={styles.detailText} />
                {jobDetails.requirements.map((requirement, index) => (
                  <T
                    key={`${index}-${requirement}`}
                    style={styles.extraLarge}
                    text={`- ${requirement}`}
                  />
                ))}
              </View>
            </View>
          )}

          <View style={styles.detailContainer}>
            <View>
              <FontAwesome5 name="user-circle" size={24} color="black" />
            </View>
            <View style={styles.ML}>
              <T id="report_to" style={styles.detailText} />
              <T
                style={styles.extraLarge}
                text={`${jobDetails.company.reportTo.name} ${
                  jobDetails.company.reportTo.phone
                    ? jobDetails.company.reportTo.phone
                    : ''
                }`}
              />
            </View>
          </View>
        </View>
      </ScrollView>
      <View style={styles.buttonContainer}>
        <ButtonComponent
          testID="no-thanks-button"
          mode="outlined"
          color={colors.off_white}
          textColor={colors.off_white}
          onPress={() => acceptRejectJobs(false, jobDetails)}
          messageId="no_thanks"
        />
        <ButtonComponent
          mode="contained"
          color={colors.black}
          textColor={colors.white}
          onPress={() => acceptRejectJobs(true, jobDetails)}
          messageId="take_it"
        />
      </View>
    </View>
  </View>
);
JobDetails.propTypes = {
  item: PropTypes.object,
  acceptRejectJobs: PropTypes.func
};
export default JobDetails;
