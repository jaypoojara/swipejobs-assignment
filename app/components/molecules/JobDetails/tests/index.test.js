import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import { renderProvider } from '../../../../utils/testUtils';
import JobDetails from '../index';

describe('<JobDetails /> Component tests', () => {
  const acceptRejectJobsSpy = jest.fn();

  const mockItem = {
    jobId: '5775d8e18a488e6c5bb08333',
    jobTitle: {
      name: 'Construction General Helper',
      imageUrl: 'https://imgs.swipejobs.com/js/job-category/construction-1.jpg'
    },
    company: {
      name: 'Steve Smith Construction',
      address: {
        formattedAddress: '430 Smith St, Chicago, IL 60654, USA',
        zoneId: 'America/Chicago'
      },
      reportTo: {
        name: 'Judy Smith',
        phone: '2130010012'
      }
    },
    wagePerHourInCents: 950,
    milesToTravel: 3.4,
    shifts: [
      {
        startDate: '2019-09-04T21:00:00Z',
        endDate: '2019-09-05T05:00:00Z'
      }
    ],
    branch: 'Downtown',
    branchPhoneNumber: '2531233322'
  };

  it('should render and match the snapshot', () => {
    const baseElement = renderProvider(<JobDetails item={mockItem} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should call acceptRejectJobs when user presses no thanks button', () => {
    const baseElement = renderProvider(
      <JobDetails item={mockItem} acceptRejectJobs={acceptRejectJobsSpy} />
    );
    fireEvent.press(baseElement.getByTestId('no-thanks-button'));
    expect(acceptRejectJobsSpy).toHaveBeenCalled();
  });
});
