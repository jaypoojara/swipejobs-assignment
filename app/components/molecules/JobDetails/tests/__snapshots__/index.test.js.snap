// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobDetails /> Component tests should render and match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eeeeee",
      "height": "85%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 8,
        "flex": 1,
        "margin": 20,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "borderRadius": 8,
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "borderRadius": 8,
              "flex": 1,
              "overflow": "hidden",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://imgs.swipejobs.com/js/job-category/construction-1.jpg",
              }
            }
            style={
              Object {
                "height": 150,
              }
            }
          />
          <View
            style={
              Object {
                "marginHorizontal": 10,
                "marginVertical": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
              testID="t"
            >
              Construction General Helper
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginTop": 10,
                }
              }
              testID="t"
            >
              Steve Smith Construction
            </Text>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#44d4ab",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "paddingVertical": 10,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 10,
                    "fontWeight": "bold",
                  }
                }
                testID="t"
              >
                Distance
              </Text>
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 25,
                    "fontWeight": "bold",
                    "marginTop": 5,
                  }
                }
                testID="t"
              >
                3.40 miles
              </Text>
            </View>
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 10,
                    "fontWeight": "bold",
                    "textAlign": "right",
                  }
                }
                testID="t"
              >
                Hourly Rate
              </Text>
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexDirection": "row",
                    "marginTop": 5,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 16,
                      "fontWeight": "bold",
                      "lineHeight": 30,
                      "textAlign": "right",
                    }
                  }
                  testID="t"
                >
                  $ 
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 25,
                      "fontWeight": "bold",
                      "textAlign": "right",
                    }
                  }
                  testID="t"
                >
                  9.50
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "marginHorizontal": 15,
                "paddingVertical": 10,
              }
            }
          >
            <View>
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "marginLeft": 15,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
                testID="t"
              >
                Shift Dates
              </Text>
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 16,
                  }
                }
                testID="t"
              >
                SEP 05, THU 02:30 AM - 10:30 AM IST
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "marginHorizontal": 15,
                "paddingVertical": 10,
              }
            }
          >
            <View>
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "marginLeft": 15,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
                testID="t"
              >
                Location
              </Text>
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 16,
                  }
                }
                testID="t"
              >
                430 Smith St, Chicago, IL 60654, USA
              </Text>
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 14,
                    "marginTop": 5,
                  }
                }
                testID="t"
              >
                3.40 miles from your job search location
              </Text>
            </View>
            <View
              style={
                Object {
                  "marginLeft": 10,
                }
              }
            >
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "marginHorizontal": 15,
                "paddingVertical": 10,
              }
            }
          >
            <View>
              <View
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "marginLeft": 15,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
                testID="t"
              >
                Report To
              </Text>
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 16,
                  }
                }
                testID="t"
              >
                Judy Smith 2130010012
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "rgba(0, 0, 0, 0.29)",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0.5,
            "elevation": 0,
            "minWidth": 64,
            "width": "45%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
          testID="no-thanks-button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {
                  "height": 50,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    false,
                    Object {
                      "color": "#acacac",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#acacac",
                  }
                }
                testID="t"
              >
                No Thanks
              </Text>
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#000000",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "minWidth": 64,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
            "width": "45%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {
                  "height": 50,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    false,
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                  }
                }
                testID="t"
              >
                I'll Take it
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
